### DISCUSSION TITLE
Troobleshoot ESP-01 Relay

### DISCUSSION THREAD

**Discussion by baptisteArno** [arendst/Tasmota#13947] [Q&A] [answered]
I'm trying to set up Tasmota with an ESP-01 Relay so that I can turn on my Christmas lights with my smartphone.

<img src="https://user-images.githubusercontent.com/16015833/145063135-16507d3b-665d-4c74-ab03-0d496c592dd0.jpg" width="200" />

<img src="https://user-images.githubusercontent.com/16015833/145063146-c54f9b59-a548-4d1c-b595-d9596fb4fd59.jpg" width="200" />

I've tried setting up both these templates:
- https://templates.blakadder.com/ESP-01S-Relay-v4.html
- https://templates.blakadder.com/ESP-01S-Relay-v1.html

But no chance, it doesn't change relay state when clicking on `Toggle`:

<img src="https://user-images.githubusercontent.com/16015833/145063002-fc7a9733-f316-4256-a18c-053b2fb83dae.gif" width="200" />

How can I debug this?

**Comment by Jason2866** [comment]
There are many boards with errors in the PCB design sold. I got myself one with three! errors.
So probably you have got such a faulty one.

**Reply by baptisteArno** [reply]
I managed to make it work with an Arduino IoT cloud project. So the Relay is working fine. But I want to make it work with Tasmota

I just don't know how to debug this

**Reply by Jason2866** [reply]
Ahh okay which gpio is used in the sketch? Configure this GPIO in Tasmota as relay.
Use module generic! Not sonoff basic

**Reply by baptisteArno** [reply]
I used GPIO 0:
```c
#define RELAY 0
...
pinMode(RELAY,OUTPUT);
digitalWrite(RELAY, LOW);
```
My current config looks ok, right?:
<img width="462" alt="CleanShot 2021-12-07 at 18 56 25@2x" src="https://user-images.githubusercontent.com/16015833/145081630-ecf24ad7-e51e-4cb2-97ae-395ace108028.png">

**Reply by baptisteArno** [reply]
Update:
When measuring the voltage on the GPIO0 pin it is constantly measuring 3.3V even though I trigger the Toggle button. It should measure 0V when the pin is "off".
How can I make sure the toggle button is supposed to change the GPIO0 pin state?

**Comment by baptisteArno** [comment, marked as answer]
I managed to make it work by importing again this template: https://templates.blakadder.com/ESP-01S-Relay-v4.html

It was my mistake I guess!
